site/docs/capabilities/usage-based-ratelimiting.md

```diff
@@ -0,0 +1,288 @@
+---
+id: usage-based-ratelimiting
+title: Usage-based Rate Limiting
+sidebar_position: 5
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+This guide will help you configure usage-based rate limiting for your AI Gateway to control token consumption across different LLM requests.
+
+## Overview
+
+Usage-based rate limiting allows you to control and monitor token consumption for your LLM requests. You can set separate limits for:
+- Input tokens
+- Output tokens
+- Total tokens
+
+This is particularly useful for:
+- Controlling costs per user
+- Implementing fair usage policies
+- Preventing abuse of your LLM endpoints
+
+## Configuration
+
+### 1. Configure Token Tracking
+
+First, you need to configure which metadata keys will store the token counts from LLM requests. Add the following configuration to your `AIGatewayRoute`:
+
+```yaml
+spec:
+  # ... other configuration ...
+  llmRequestCosts:
```

Comment:
It also supports `CELExpression` using a custom formula to calculate the cost based in input/output token.